### DISCUSSION TITLE
Validating Input with additional fields

### DISCUSSION THREAD

**Discussion by blackjack4494** [fastapi/fastapi#9680] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
# models
class Player(Base):
    __tablename__ = "players"

    id = Column(Integer, autoincrement=True, primary_key=True, unique=True, index=True)
    uuid = Column(String(36), index=True)
    account_id = Column(String(64), unique=True, index=True)
    name = Column(String(64))
    instance_id = Column(Integer, nullable=True)

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.uuid = str(...)

# schemas
class PlayerBase(BaseModel):
    account_id: str
    name: str
    instance_id: Union[int, None] = -1


class PlayerCreate(PlayerBase):
    pass


class Player(PlayerBase):
    id: int
    uuid: str

    class Config:
        orm_mode = True
        schema_extra = {
            "example": {...}
        }

# crud
async def create_player(db: AsyncSession, player: schemas.PlayerCreate):
    db_player = models.Player(**player.dict())
    db.add(db_player)
    await db.commit()
    await db.refresh(db_player)
    return db_player

# main/api
@app.post("/players/", response_model=schemas.Player, responses=responses)
async def create_player(player: schemas.PlayerCreate, db: Session = Depends(get_db)):
    db_player = await crud.get_player_by_account_id(db, account_id=player.account_id)
    if db_player:
        raise CustomException(status_code=409, message="Player already exists")
    ret = await crud.create_player(db=db, player=player)
    return ret
```

### Description

My code works fine for its intended use case with this json as input:  

```json
{
  "account_id": "test123",
  "name": "name123",
  "instance_id": -1
}
```

Now if I change it to the following

```json
{
  "account_id": "test123",
  "name": "name123",
  "instance_id": -1,
  "random_key": "random_value"
}
```

It is accepted as well. However I don't want it to be accepted.  
I tried using `{schema}.from_orm` but that didn't really work.  
There is probably already a solution to this but I am unable to find it or am lacking the proper wording to describe it (sorry for that).  

Bonus:
```json
{
  "account_id": "test123",
  "name": "name123",
}
```

Note the additional comma.
This body will be rejected with `422 Unprocessable Entity`. With the comma removed it works fine again.  
Is there a way to allow this somehow?

I appreciate the help :)
If there are unrelated suggestions I am open to them as well.

### Operating System

Windows

### FastAPI Version

0.88.0

### Python Version

3.9.0

**Comment by jgould22** [comment, marked as answer]
Does setting "extra" to forbid in your model config help at all?

https://docs.pydantic.dev/latest/usage/model_config/#options

**Reply by blackjack4494** [reply]
Indeed. I somehow missed that.  
My solution is:
```python
class BaseModelRestrictive(BaseModel):
    class Config:
        extra = "forbid"

class PlayerCreate(PlayerBase, BaseModelRestrictive):
  ...
```

Thank you.
